stablehlo/reference/Element.h

```diff
@@ -79,6 +98,13 @@ inline raw_ostream &operator<<(raw_ostream &os, Element element) {
   return os;
 }
 
+/// Check if the type 'type' comforms with what is supported in StableHLO.
+bool isSupportedUnsignedIntegerType(Type type);
+bool isSupportedSignedIntegerType(Type type);
+bool isSupportedIntegerType(Type type);
+bool isSupportedFloatType(Type type);
+bool isSupportedComplexType(Type type);
```

Comment:
Originally, we discussed putting these functions in the `mlir::stablehlo::detail` namespace similarly to where we put `Buffer` (since they aren't e.g. supposed to be used in Ops.cpp). What do you think about that?